impeller/renderer/backend/vulkan/allocator_vk.cc

```diff
@@ -357,61 +387,36 @@ class AllocatedTextureSourceVK final : public TextureSourceVK {
                      << vk::to_string(result);
       return;
     }
-    resource_.Reset(
-        ImageResource(image, allocator, allocation, std::move(image_view)));
+    resource_.Reset(ImageResource(ImageVMA{allocator, allocation, image},
+                                  std::move(image_view)));
     is_valid_ = true;
   }
 
   ~AllocatedTextureSourceVK() = default;
 
   bool IsValid() const { return is_valid_; }
 
-  vk::Image GetImage() const override { return resource_->image; }
+  vk::Image GetImage() const override { return resource_->image.get().image; }
 
   vk::ImageView GetImageView() const override {
     return resource_->image_view.get();
   }
 
  private:
   struct ImageResource {
-    vk::Image image = {};
-    VmaAllocator allocator = {};
-    VmaAllocation allocation = {};
+    UniqueImageVMA image;
     vk::UniqueImageView image_view;
 
     ImageResource() = default;
 
-    ImageResource(vk::Image p_image,
-                  VmaAllocator p_allocator,
-                  VmaAllocation p_allocation,
-                  vk::UniqueImageView p_image_view)
-        : image(p_image),
-          allocator(p_allocator),
-          allocation(p_allocation),
-          image_view(std::move(p_image_view)) {}
+    ImageResource(ImageVMA p_image, vk::UniqueImageView p_image_view)
+        : image(p_image), image_view(std::move(p_image_view)) {}
 
     ImageResource(ImageResource&& o) {
       std::swap(image, o.image);
-      std::swap(allocator, o.allocator);
-      std::swap(allocation, o.allocation);
       std::swap(image_view, o.image_view);
     }
 
-    ~ImageResource() {
-      if (!image) {
-        return;
-      }
-      TRACE_EVENT0("impeller", "DestroyDeviceTexture");
-      image_view.reset();
```

Comment:
The image view is already a unique object. It had to be reset manually earlier because the image was explicitly collected. With the RAII wrapper, there is no need as the ivar destruction order holds.